app/sessionStore.js

```diff
@@ -421,64 +514,30 @@ module.exports.loadAppState = () => {
       data = fs.readFileSync(getStoragePath())
     } catch (e) {}
 
+    let loaded = false
     try {
       data = JSON.parse(data)
-      // autofill data migration
-      if (data.autofill) {
-        if (Array.isArray(data.autofill.addresses)) {
-          let addresses = exports.defaultAppState().autofill.addresses
-          data.autofill.addresses.forEach((guid) => {
-            addresses.guid.push(guid)
-            addresses.timestamp = new Date().getTime()
-          })
-          data.autofill.addresses = addresses
-        }
-        if (Array.isArray(data.autofill.creditCards)) {
-          let creditCards = exports.defaultAppState().autofill.creditCards
-          data.autofill.creditCards.forEach((guid) => {
-            creditCards.guid.push(guid)
-            creditCards.timestamp = new Date().getTime()
-          })
-          data.autofill.creditCards = creditCards
-        }
-        if (data.autofill.addresses.guid) {
-          let guids = []
-          data.autofill.addresses.guid.forEach((guid) => {
-            if (typeof guid === 'object') {
-              guids.push(guid['persist:default'])
-            } else {
-              guids.push(guid)
-            }
-          })
-          data.autofill.addresses.guid = guids
-        }
-        if (data.autofill.creditCards.guid) {
-          let guids = []
-          data.autofill.creditCards.guid.forEach((guid) => {
-            if (typeof guid === 'object') {
-              guids.push(guid['persist:default'])
-            } else {
-              guids.push(guid)
-            }
-          })
-          data.autofill.creditCards.guid = guids
-        }
-      }
-      // xml migration
-      if (data.settings) {
-        if (data.settings[settings.DEFAULT_SEARCH_ENGINE] === 'content/search/google.xml') {
-          data.settings[settings.DEFAULT_SEARCH_ENGINE] = 'Google'
-        }
-        if (data.settings[settings.DEFAULT_SEARCH_ENGINE] === 'content/search/duckduckgo.xml') {
-          data.settings[settings.DEFAULT_SEARCH_ENGINE] = 'DuckDuckGo'
-        }
+      loaded = true
+    } catch (e) {
+      // Session state might be corrupted; let's backup this
+      // corrupted value for people to report into support.
+      module.exports.backupSession()
```

Comment:
There's more to do here as per discussions but doesn't need to all be done in this bug.
But I think this will try to do a sync copy on disk on first startup which would slow down startup. Could you verify if that's the case?